core/modules/taxonomy/taxonomy.module

```diff
@@ -1852,6 +1857,32 @@ function taxonomy_field_widget_form(&$form, &$form_state, $field, $instance, $la
   return $element;
 }
 
+/**
+ * Implements hook_field_widget_form_alter().
+ */
+function taxonomy_field_widget_form_alter(&$element, &$form_state, $context) {
+  if ($context['field']['type'] == 'taxonomy_term_reference' && in_array($element['#type'], array(
+    'checkboxes',
+    'radios',
+  ))) {
+    $option_extras = array();
+    foreach ($context['field']['settings']['allowed_values'] as $tree) {
+      if ($vocabulary = taxonomy_vocabulary_load($tree['vocabulary'])) {
+        if ($terms = taxonomy_get_tree($vocabulary->machine_name, $tree['parent'])) {
+          foreach ($terms as $term) {
+            $option_extras[$term->tid] = array('#attributes' => array('class' => array('taxonomy-term-depth-' . $term->depth)));
+          }
+        }
+      }
+    }
+    $element += $option_extras;
+    $element['#attached'] = array(
```

Comment:
I suggest you don't attach css this way. This will "wipe out" all other css attachments done by other widget alter hooks.

Instead I suggest you use:

`$element['#attached']['css'][] = backdrop_get_path('module', 'taxonomy') . '/css/taxonomy.css';`

Reply:
Yes, thanks!